feat: add docs for GitHub secret scanning program (#3049)
buildkite/docs: pages/apis.md

All webhooks contain an [`X-Buildkite-Token` header](/docs/apis/webhooks#http-headers) which allows you to verify the authenticity of the request.

## API security

> We are a member of the [GitHub Secret Scanning Program](https://docs.github.com/en/code-security/secret-scanning/secret-scanning-partnership-program/secret-scanning-partner-program).

Reviewer: ```suggestion
Buildkite is a member of the [GitHub Secret Scanning Program](https://docs.github.com/en/code-security/secret-scanning/secret-scanning-partnership-program/secret-scanning-partner-program).
```

Reviewer: See the section on [Formality](https://github.com/buildkite/docs/blob/main/styleguides/writing-style.md#formality) in the Writing style guide.

This service [alerts](https://docs.github.com/en/code-security/secret-scanning/secret-scanning-partnership-program/secret-scanning-partner-program#the-secret-scanning-process) us when a Buildkite API key has been leaked on GitHub in a public repository.

> Once we receive a notification of a publicly leaked token from GitHub, we will:

Reviewer: ```suggestion
Once Buildkite receives a notification of a publicly leaked token from GitHub, Buildkite will:
```

> - Revoke the token immediately

Reviewer: ```suggestion
- Revoke the token immediately.
```

- Email the user who generated the token to let them know it has been revoked.
- Email the organizations associated with the token to let them know it has been revoked.

> This can also be enabled for [private repositories](https://docs.github.com/en/code-security/secret-scanning/enabling-secret-scanning-features/enabling-secret-scanning-for-your-repository).

Reviewer: ```suggestion
You can also enable GitHub secret scanning for [private repositories](https://docs.github.com/en/code-security/secret-scanning/enabling-secret-scanning-features/enabling-secret-scanning-for-your-repository).
```

> This can also be enabled for [private repositories](https://docs.github.com/en/code-security/secret-scanning/enabling-secret-scanning-features/enabling-secret-scanning-for-your-repository).
> 
> You can generate a new access token [here](https://buildkite.com/user/api-access-tokens).

Reviewer: ```suggestion
You can also:

- Enable GitHub secret scanning for [private repositories](https://docs.github.com/en/code-security/secret-scanning/enabling-secret-scanning-features/enabling-secret-scanning-for-your-repository).

- Generate a new [access token for your Buildkite user account](https://buildkite.com/user/api-access-tokens).
```

Reviewer: Last suggestion to commit ^ @jfhinchcliffe and then this is ready to merge in!

Reviewer: Sorry, @jfhinchcliffe , I ended up committing this one myself, as I suspect you would've be cool with this.

All webhooks contain an [`X-Buildkite-Token` header](/docs/apis/webhooks#http-headers) which allows you to verify the authenticity of the request.

## API security

> Buildkite is a member of the [GitHub Secret Scanning Program](https://docs.github.com/en/code-security/secret-scanning/secret-scanning-partnership-program/secret-scanning-partner-program).

Reviewer: ```suggestion
Buildkite is a member of the [GitHub secret scanning program](https://docs.github.com/en/code-security/secret-scanning/secret-scanning-partnership-program/secret-scanning-partner-program).
```

Reviewer: Product features should not use Title Case:
https://github.com/buildkite/docs/blob/main/styleguides/writing-style.md#product-names-and-product-features

This service [alerts](https://docs.github.com/en/code-security/secret-scanning/secret-scanning-partnership-program/secret-scanning-partner-program#the-secret-scanning-process) us when a Buildkite API key has been leaked on GitHub in a public repository.

> Once Buildkite receives a notification of a publicly leaked token from GitHub, Buildkite will:
> - Revoke the token immediately.

Reviewer: ```suggestion
Once Buildkite receives a notification of a publicly leaked token from GitHub, Buildkite will:

- Revoke the token immediately.
```

Reviewer: Add line break to make bullet list render properly in HTML output.
